powertools-core/src/main/java/software/amazon/lambda/powertools/core/internal/LambdaHandlerProcessor.java

```diff
@@ -78,4 +78,8 @@ public static boolean isColdStart() {
     public static void coldStartDone() {
         IS_COLD_START = false;
     }
+
+    public static boolean isSamLocal() {
+        return "true".equals(System.getenv("AWS_SAM_LOCAL"));
```

Comment:
Is this change supposed to be part of this pull request?

Reply:
Well it could be different PR as well, but I just made a separate commit to same PR. But yeah its intentional.